Rm/lr state request

ricardomilhazes opened the pull request.

ricardomilhazes committed:
Added method to retrieve loan requests at a specific state

ricardomilhazes committed:
Added method to retrieve loan requests at a specific state

ricardomilhazes committed:
Added method to retrieve loan requests at a specific state

ricardomilhazes committed:
Added method to retrieve loan requests at a specific state

ricardomilhazes committed:
Corrected minor mistake

ricardofsc10 reviewed (commented):
Nice job mate.

albertofaria committed:
Set up GitHub Actions and black (#41)

* Add black configuration

* Reformat code using black

* Add GitHub Actions setup

hpereira98 commented on tuichain/api/views/loanrequests.py:
```diff
@@ -165,6 +165,26 @@ def get_non_validated_loan_requests(request):
 
+@api_view(["GET"])
+@permission_classes((IsAdminUser,))
+def get_specific_state_loan_requests(request, status):
```
I think every user should be able to do this, with some restrictions:
- pending, cancelled and finished and expired can only be seen by the user that requested the loans
- active should be seen by the users that requested it or invested on it
- funding can be seen by everyone

ricardomilhazes committed:
Updated method permissions

ricardomilhazes committed:
Updated method permissions

ricardomilhazes committed:
Updated method permissions

ricardomilhazes committed:
Updated method permissions

ricardomilhazes committed:
Completed all requested methods

ricardomilhazes committed:
Completed all requested methods

ricardomilhazes committed:
Completed all requested methods

ricardomilhazes committed:
Merge branch 'rm/lr_state_request' of https://github.com/TuiChain/backend into rm/lr_state_request

ricardomilhazes committed:
Formatted files with black

ricardomilhazes committed:
Formatted files with black

albertofaria committed:
Install blockchain component in Actions setup (#44)

This is necessary since this dependency won't be included in the
requirements.txt file.

ricardomilhazes committed:
Rm/update models (#42)

* Updated models and also updated all methods that needed to be changed

* Updated to_dict methods

* Set up GitHub Actions and black (#41)

* Add black configuration

* Reformat code using black

* Add GitHub Actions setup

* Made corrections pointed out on previous pr, still needs testing

* Updated models and also updated all methods that needed to be changed

* Updated to_dict methods

* Made corrections pointed out on previous pr, still needs testing

* Minor changes

* Formatted files using black

Co-authored-by: Alberto Faria <albertofaria.dev@gmail.com>

ricardomilhazes committed:
Merge branch 'rm/lr_state_request' of https://github.com/TuiChain/backend into rm/lr_state_request

ricardomilhazes committed:
Corrected minor mistakes
